repos/system_upgrade/common/actors/setuptargetrepos/tests/test_repomapping.py

```diff
@@ -0,0 +1,566 @@
+        'and have the major version same as the source system.')
+    expected_pesid_repos = [repositories[0], repositories[1]]
+    actual_pesid_repos = handler.get_source_pesid_repos('pesid1')
+    assert len(expected_pesid_repos) == len(actual_pesid_repos), fail_description
+    for actual_pesid_repo in actual_pesid_repos:
+        assert actual_pesid_repo in expected_pesid_repos, fail_description
+
+    fail_description = (
+        'The get_source_pesid_repos method does not take into account the source system version correctly.'
+    )
+    monkeypatch.setattr(setuptargetrepos_repomap, 'get_source_major_version', lambda: '10')
+
+    # Repeat the same test as above to make sure it respects the source OS major version
+    assert [] == handler.get_source_pesid_repos('pesid1'), fail_description
+
+    assert [] == handler.get_source_pesid_repos('pesid2'), fail_description
+    assert [] == handler.get_source_pesid_repos('nonexisting_pesid'), fail_description
+
+
+def test_get_target_pesid_repos(monkeypatch, repomap_data_for_pesid_repo_retrieval):
+    """
+    Test for the RepoMapDataHandler.get_target_pesid_repos method.
+
+    Verifies that the method is able to collect all PESIDRepositoryEntry that match the given PES ID and
+    have the major version same as the source system.
+    """
+    monkeypatch.setattr(setuptargetrepos_repomap, 'get_target_major_version', lambda: '8')
+    handler = RepoMapDataHandler(repomap_data_for_pesid_repo_retrieval)
+    repositories = repomap_data_for_pesid_repo_retrieval.repositories
+
+    fail_description = (
+        'The get_target_pesid_repos method failed to retrieve all pesid repos that match given pesid '
+        'and have the major version same as the target system.')
+    expected_pesid_repos = [repositories[3], repositories[4], repositories[5]]
+    actual_pesid_repos = handler.get_target_pesid_repos('pesid3')
+    assert len(expected_pesid_repos) == len(actual_pesid_repos), fail_description
+    for actual_pesid_repo in actual_pesid_repos:
+        assert actual_pesid_repo in expected_pesid_repos, fail_description
+
+    fail_description = (
+        'The get_target_pesid_repos method doesn\'t take into account the taget system version correctly.'
+    )
+    monkeypatch.setattr(setuptargetrepos_repomap, 'get_target_major_version', lambda: '10')
+
+    # Repeat the same test as above to make sure it respects the target OS major version
+    assert [] == handler.get_target_pesid_repos('pesid3'), fail_description
+
+    assert [] == handler.get_target_pesid_repos('pesid1'), fail_description
+    assert [] == handler.get_target_pesid_repos('nonexisting_pesid'), fail_description
+
+
+@pytest.fixture
+def mapping_data_for_find_repository_equiv():
+    repositories = [
+        make_pesid_repo('pesid1', '7', 'pesid1-repoid'),
+        make_pesid_repo('pesid1', '7', 'pesid1-repoid', channel='e4s', rhui='aws'),
```

Comment:
This is good case to think about. However in this case I am afraid the result is based on the order of these two repositories. I am thinking whether to keep it as it is or whether to change the second repoid.

Reply:
Oh, I've missed the fact that I put there the same `repoid` twice (it was not intended). I will change it so that the `repoid` is unique, because I don't think it is a good idea that the result of an unit test is based on some "hidden knowledge" (e.g.  the fact that the repository matters).

Reply:
The non-unique repoid is fixed.